callisto/delivery/models.py

```diff
@@ -157,7 +168,31 @@ def decrypted_report(self, key):
         Raises:
           CryptoError: If the key and saved salt fail to decrypt the record.
         """
-        return _decrypt_report(salt=self.salt, key=key, encrypted=self.encrypted)
+        iterations = None
+        harden_runtime = False
+        hasher = identify_hasher(self.encode_prefix)
+
+        if self.encode_prefix == '':
```

Comment:
ok, sounds like jamming it into must_update doesn't make sense. I think I'd still like to see the lack-of-prefix check happen on the hasher rather than in the model, but I'm open to argument here.